src/CSET/operators/read.py

```diff
@@ -136,15 +136,38 @@ def read_cubes(
     FileNotFoundError
         If the provided path does not exist
     """
-    input_files = _check_input_files(loadpath, filename_pattern)
-    # A constraint of None lets everything be loaded.
     logging.debug("Constraint: %s", constraint)
+
+    def load_from_paths(
+        paths: list[str], is_ensemble: bool, is_base: bool
+    ) -> iris.cube.CubeList:
+        # Skip if no paths given, mainly for when we only have a base path.
+        if not paths:
+            return iris.cube.CubeList([])
+        input_files = _check_input_files(paths, filename_pattern)
+        callback = _create_callback(is_ensemble=is_ensemble, is_base=is_base)
+        # If unset, a constraint of None lets everything be loaded.
+        return iris.load(input_files, constraint, callback=callback)
```

Comment:
```suggestion
        return iris.load(input_files, constraint=None, callback=callback)
```
Realise in comment but for readibility more explicit? Probably personal preference.

Reply:
Here `constraint` is referring to a variable passed in as an argument to the operator. This variable may be None, but it may also contain a constraint.

This change would always set the constraint to None, so we would never constrain regardless of what the constraint argument is set to. The actual default comes from the `def read_cubes` line, where we do have `constraint=None`.

Reply:
Doh, misread this as the function definition.